sql/src/main/java/org/apache/druid/sql/calcite/external/ExternalTableScanRule.java

```diff
@@ -46,6 +46,7 @@ public boolean matches(RelOptRuleCall call)
     if (plannerContext.getQueryMaker().feature(QueryFeature.CAN_READ_EXTERNAL_DATA)) {
       return super.matches(call);
     } else {
+      plannerContext.setPlanningError("Scanning external datasources is not suported.");
```

Comment:
Typically since the planning errors are hints, we follow this convention 
Instead of "Feature X is not supported", we say "SQL query requires feature X that is not supported." The reason for this is that the actual error user will see. This is what they see
"Possible error: SQL query requires feature X that is not supported" 
we prepend "Possible error: " to all the planning error messages. When the error is phrased as "SQL query requires.. ", we are being explicit that we think SQL query being executed needs feature X but can't say for sure. Though it is for certain that feature X is not supported. It's a bit of a nitpick but since the error message is seen by the end-user, we should address it.

Reply:
Thanks for the explanation. I will make the changes so that it points the user towards the correct error, rather than being definitive (as it might be incorrect).